### PR TITLE
stops pilots breaking the bank

### PR DESCRIPTION
**Pull request by harryob**
:cl:
fix: pilots can no longer take out enormous loans to buy their equipment
/:cl:

### CODE REVIEW

File: code/modules/cm_marines/vehicle_part_fabricator.dm [hunk 1]
```diff
@@ -149,7 +157,7 @@
 		var/index = params["index"]
```
Comment (Drulikar) [inline code/modules/cm_marines/vehicle_part_fabricator.dm:157]: I doubt this is bug caused by this PR (more likely #7084 ) ; but in testing this I am able to get the index == null if I queue up items and cancel them: 
image

Seems fairly easy to reproduce by queueing say 5 things, and then spam clicking cancel on whatever is item 2 until you get an exception.

**Review by Drulikar — DISMISSED** [review]

File: code/modules/cm_marines/vehicle_part_fabricator.dm [hunk 2]
```diff
@@ -149,7 +157,7 @@
...
 
 		if(length(build_queue))
-			if(index == 1)
+			if(busy && index == 1)
 				to_chat(user, SPAN_WARNING("Cannot cancel currently produced item."))
 				return
```
Comment (Sindorman) [inline code/modules/cm_marines/vehicle_part_fabricator.dm:163]: This to fix issue @Drulikar was seeing
```suggestion
				if(index == null || index > length(build_queue))
					return
```

**Review by realforest2001 — APPROVED** [review]

File: code/modules/cm_marines/vehicle_part_fabricator.dm [hunk 3]
```diff
@@ -143,21 +151,24 @@
...
 			build_part(produce, cost, user)
-			return
+			return TRUE
 
 	if(action == "cancel")
 		var/index = params["index"]
 
 		if(length(build_queue))
-			if(index == 1)
+			if(index > length(build_queue))
```
Comment (Sindorman) [inline code/modules/cm_marines/vehicle_part_fabricator.dm:160]: This should still have a `null` check since @Drulikar was able to get `index = null` edge case
```suggestion
			if(index == null || index > length(build_queue))
```

Comment (harryob) [reply code/modules/cm_marines/vehicle_part_fabricator.dm:160]: the error in drathek's screenshot is not null, it's going oob because the tgui value had not updated in the act. but sure

**Review by Nanu308 — APPROVED** [review]
awesum
